docs/parse.md

```diff
@@ -0,0 +1,47 @@
+# `Temporal.parse`
+
+The `Temporal.parse` function parses a full ISO string and divides it into its
+respective components, which can then be used to produce the corresponding
+Temporal objects.
+
+This function can be used to parse full valid ISO strings that represent invalid
+absolutes and salvage valid components from the string.
+
+## Syntax
+
+```javascript
+const parseResult = Temporal.parse(input);
+```
+
+### Parameters
+
+- `input` (`string`): The input string to be parsed. It should be a full valid
+ISO string, but it may or may not represent a valid absolute.
+
+### Returns
+
+The returned value is an `object` that contains all the components of the parsed
+ISO string. It has a structure described by the following TypeScript interface:
+
+```typescript
+interface parseResult {
```

Comment:
I have mixed feelings about this. On the one hand, it is very clear and concise and expresses exactly what is meant. On the other hand, before I knew any TypeScript, I found it annoying when documentation did this; after all, I'm writing JavaScript, not TypeScript.

If you decide to put the signature of the returned object in the h3 as suggested above, then I'd say this isn't necessary.

That said, I'd either make these all optional, or make them `string | null` (my preference), because we need a way to represent what happens when you are parsing a partial string.

Reply:
> I have mixed feelings about this.

I share your sentiments exactly. I tried to come up with a different way to document this which was just as clear yet concise, but couldn't. I'd understand if you'd ask me to do it through text instead.

> because we need a way to represent what happens when you are parsing a partial string

As of right now, we're not accepting partial ISO strings, nor was I planning to change that, since I didn't get that sentiment so far. Do you think we should support partial ISO strings?

Reply:
Oh, I hadn't realized that yet. I do think we need this. For example, DateTime strings without a time zone can just as well have invalid parts and need to be split up.

In the [draft doc](http://tc39.es/proposal-temporal/docs/parse-draft.html) I suggested something like this:
```js
Temporal.parse('2001-09-09T01:46:40')
// {
//   absolute: null,
//   dateTime: '2001-09-09T01:46:40',
//   date: '2001-09-09',
//   yearMonth: '2001-09',
//   monthDay: '09-09',
//   time: '01:46:40',
//   offset: null,
//   timeZone: null,
// }
```

Reply:
apart from the optional time zone, do you think there's any cases you'd want to support? Like parsing just a `Date` to get a `Date`, `YearMonth` and `MonthDay`? Asking this since I'd need to go back to my regex changes and undo some of the refactoring I did around `ES.ParseISODateTime`.

Reply:
> parsing just a `Date` to get a `Date`, `YearMonth` and `MonthDay`?

That was one of the use cases that I mentioned in http://tc39.es/proposal-temporal/docs/parse-draft.html, as well. Supporting it does make sense to me, but I'm open to being convinced otherwise.

Reply:
I'm sorry I missed that case. No, I don't want to convince you otherwise, I just wanted to ask your preference since my implementation would depend on that.